pkg/textparse/openmetricsparse.go

```diff
@@ -85,10 +87,19 @@ type OpenMetricsParser struct {
 	hasTS   bool
 	start   int
 	offsets []int
+
+	eOffsets      []int
+	exemplar      []byte
+	exemplarVal   float64
+	exemplarTs    int64
+	hasExemplarTs bool
 }
 
 // NewOpenMetricsParser returns a new parser of the byte slice.
 func NewOpenMetricsParser(b []byte) Parser {
+	if len(b) > 0 && b[len(b)-1] != '\n' {
+		b = append(b, '\n')
```

Comment:
Ideally we wouldn't need to append an extra endline - can we add this support without this requirement?

If it adds an extra character the entire buffer might need re-allocation, also not guaranteed that this is safe to do either if it's a subslice of a bigger slice passed to this function.